presto-hive/src/main/java/com/facebook/presto/hive/util/HiveFileIterator.java

```diff
@@ -92,14 +96,14 @@ protected HiveFileInfo computeNext()
             if (paths.isEmpty()) {
                 return endOfData();
             }
-            remoteIterator = getLocatedFileStatusRemoteIterator(paths.removeFirst());
+            remoteIterator = getLocatedFileStatusRemoteIterator(paths.removeFirst(), pathFilter);
         }
     }
 
-    private Iterator<HiveFileInfo> getLocatedFileStatusRemoteIterator(Path path)
+    private Iterator<HiveFileInfo> getLocatedFileStatusRemoteIterator(Path path, PathFilter pathFilter)
     {
         try (TimeStat.BlockTimer ignored = namenodeStats.getListLocatedStatus().time()) {
-            return new FileStatusIterator(path, listDirectoryOperation, namenodeStats);
+            return new FileStatusIterator(path, listDirectoryOperation, namenodeStats, pathFilter);
```

Comment:
nit: Did you consider using the `com.google.common.collect.Iterators#filter` instead?

Reply:
Fixed